### PR TITLE
doc: updating style guide

### PR DESCRIPTION
**Pull request by wiba-nordic**
NCSDK-9708

Improve readability, remove extraneous examples

Some matters were moved over to the cheat sheet.

Signed-off-by: Wille Backman <wille.backman@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/doc_styleguide.rst [hunk 1]
```diff
@@ -455,20 +225,20 @@ Functions
...
    There is usually only one return value.
 
    .. code-block:: none
 
       *  @return  Initializer that sets up the pipe, length, and byte array for
       *           content of the TX data.
 
-#. **@retval** should be used for specific return values (for example, "@retval true", "@retval CONN_ERROR").
+#. ``@retval`` should be used for specific return values (for example, "@retval true", "@retval CONN_ERROR").
    Describe the condition for each of the return values (for example, "If the function completes successfully", "If the connection cannot be established").
```
Comment (wiba-nordic) [inline doc/nrf/doc_styleguide.rst:242]: I think these two, and the bullet point following these (lines 481-501/251-271), could be combined to one bullet point how to format the return, with the current ``@return`` and ``@retval`` as sub-bullets for it.

File: doc/nrf/doc_styleguide.rst [hunk 2]
```diff
@@ -11,218 +11,75 @@ Documentation |sg|
...
-        hscale = "1.3";
-        Module,Application;
-        Module<<Application      [label="nrf_cloud_connect() returns successfully"];
-        Module>>Application      [label="NRF_CLOUD_EVT_TRANSPORT_CONNECTED"];
-        Module>>Application      [label="NRF_CLOUD_EVT_USER_ASSOCIATION_REQUEST"];
-        Module<<Application      [label="nrf_cloud_user_associate()"];
-        Module>>Application      [label="NRF_CLOUD_EVT_USER_ASSOCIATED"];
-        Module>>Application      [label="NRF_CLOUD_EVT_READY"];
-        Module>>Application      [label="NRF_CLOUD_EVT_TRANSPORT_DISCONNECTED"];
+Each link should be defined only once in :file:`links.txt`, so if you are adding a new link, make sure it hasn't already been added.
```
Comment (peknis) [inline doc/nrf/doc_styleguide.rst:63]:
```suggestion
Each link should be defined only once in :file:`links.txt`, so if you are adding a new link, make sure it has not already been added.
```

File: doc/nrf/doc_styleguide.rst [hunk 3]
```diff
@@ -231,6 +88,7 @@ The Breathe Sphinx plugin provides a bridge between RST and doxygen.
 
 The doxygen documentation is not automatically included in RST.
 Therefore, every group must be explicitly added to an RST file.
+The `Breathe documentation`_ contains information about what you can link to.
```
Comment (peknis) [inline doc/nrf/doc_styleguide.rst:91]: You might want to say something about the example below.

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by greg-fer — APPROVED** [review]

**Review by peknis — APPROVED** [review]

File: doc/nrf/doc_styleguide.rst [hunk 4]
```diff
@@ -362,143 +133,143 @@ General documentation guidelines
...
 
-#. If you include more than one **@sa** ("see also", optional), add them this way::
+#. If you include more than one ``@sa`` ("see also", optional), add them this way::
 
       @sa first_function
       @sa second_function
 
-#. **@return** should be used to describe a generic return value without a specific value (for example, "@return The length of ...", "@return The handle").
-   There is usually only one return value.
+#. Do not user ``@returns``, use ``return`` or ``retval`` instead.
```
Comment (peknis) [inline doc/nrf/doc_styleguide.rst:234]: This needs a blank line before the bullet.

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by peknis — APPROVED** [review]

**Pull request merged**